### ISSUE TITLE
Bug: Issue with Arel scope passed to pagy method

### ISSUE THREAD

**Issue by elliotlarson** [ddnexus/pagy#645]
### Before submitting...

- [X] I upgraded to the latest version of pagy
- [X] I searched through the [Documentation](https://ddnexus.github.io/pagy/)
- [X] I searched through the [Issues](https://github.com/ddnexus/pagy/issues)
- [X] I searched through the [Q&A](https://github.com/ddnexus/pagy/discussions/categories/q-a)

### I am providing...

- [X] Simple step by step list that would work in IRB with the [Pagy::Console](https://ddnexus.github
.io/pagy/docs/api/console) without invoking any external code

- [ ] Plain ruby file that can run as `ruby my-problem.rb`

- [ ] Edited copy of the single file [pagy_standalone_app.ru](https://github.com/ddnexus/pagy/blob/master/apps/pagy_standalone_app.ru)

- [ ] Edited copy of the single file [pagy_bare_rails.rb](https://github.com/ddnexus/pagy/blob/master/apps/pagy_bare_rails.rb)

- [ ] Link of my own branch forked from pagy, which contains an added test file

- [ ] Link of my own branch forked from [any of the rails apps listed here.](https://github.com/stars/benkoshy/lists/rails-demo-apps-for-pagy)

- [ ] A `docker-compose` file that uses only docker images and no local context and runs with `docker-compose up`

### Description

In our system we have Manufacturer, Panel, and Project models.  Panels belong to a project and a manufacturer.  On our manufacturer model we have a scope that uses an Arel statement, which appears to be causing an issue with the latest version of Pagy's pagy method.  

Here's a simplified version of the scope on the Manufacturer model:

```ruby
  scope(
    :with_counts, lambda do
      table = arel_table
      panels = Panel.arel_table
      active_panels_count = panels
        .project(Arel.star.count)
        .where(table[:id].eq(panels[:manufacturer_id]))
        .where(panels[:retired_at].eq(nil))
        .as("active_panels_count")
      select(
        table[Arel.star],
        active_panels_count,
      )
    end
  )
```

This results in the following SQL:

```sql
SELECT "manufacturers".*, (SELECT COUNT(*) FROM "panels" WHERE "manufacturers"."id" = "panels"."manufacturer_id" AND "panels"."retired_at" IS NULL) active_panels_count FROM "manufacturers"
```

We're using Pagy with the Ransack gem.  The error we're seeing is happening in the controller:

```ruby
@search = Manufacturer.with_counts.ransack(search_params)
@pagy, @manufacturers = pagy(@search.result)
```

This worked okay with the v6 version of Pagy, but now it results in this error:

```text
PG::SyntaxError: ERROR: syntax error at or near "Arel" LINE 1: SELECT COUNT(#<struct Arel::Attributes::Attribute relation=#... ^
```

**Comment by ddnexus** [comment]
Hi @elliotlarson, I would need a self contained support to reproduce this with one click and start working on the fix.
I am afraid a verbal description of your code, is not enough, sorry. Thanks.

**Comment by elliotlarson** [comment]
Okay, how about this script:

```ruby
require "bundler/inline"

gemfile do
  source "https://rubygems.org"
  gem "sqlite3"
  gem "activerecord"
  # gem "activesupport"
  gem "pagy"
  # gem "ransack"
  gem "debug"
end

require "active_record"
# require "active_support/all"

include Pagy::Backend

ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Base.establish_connection(
  adapter: "sqlite3",
  database: ":memory:", # using in memory data store
)

ActiveRecord::Schema.define do
  create_table :manufacturers do |t|
    t.string :name
  end

  create_table :panels do |t|
    t.string :name
    t.integer :manufacturer_id
  end

  create_table :projects do |t|
    t.string :name
    t.integer :panel_id
  end
end

class Manufacturer < ActiveRecord::Base
  has_many :panels

  scope(
    :with_counts, lambda do
      table = arel_table
      panels = Panel.arel_table
      active_panels_count = panels
        .project(Arel.star.count)
        .where(table[:id].eq(panels[:manufacturer_id]))
        .as("active_panels_count")
      select(
        table[Arel.star],
        active_panels_count,
      )
    end
  )
end

class Project < ActiveRecord::Base
  belongs_to :panel
  belongs_to :manufacturer
end

class Panel < ActiveRecord::Base
  belongs_to :manufacturer
end

manufacturers = Manufacturer.with_counts
debugger

pagy, manufacturers = pagy(Manufacturer.with_counts)
debugger
```

You get to the first debugger successfully, but if you continue, SQLite throws an error.

**Comment by ddnexus** [comment]
Thank you! Fixed in 7.0.2.
